tests/integration/api/test_api.py

```diff
@@ -521,221 +357,156 @@ def _compare_screenboard(apiBoard, expectedBoard):
-        monitor = dog.Monitor.get(updated_monitor_id)
-        eq(monitor['query'], query2)
+        monitor = dog.Monitor.update(monitor["id"], query=query2, options=options)
+        assert monitor["query"] == query2
+        assert monitor["options"]["notify_no_data"] == options["notify_no_data"]
+        assert monitor["options"]["silenced"] == options["silenced"]
+
+        name = "test_monitors"
+        monitor = dog.Monitor.update(monitor["id"], query=query2, name=name, options={"notify_no_data": True})
+        assert monitor["name"] == name
+        assert monitor["query"] == query2
+        assert monitor["options"]["notify_no_data"] is True
 
-        name = 'test_monitors'
-        monitor_id = dog.Monitor.update(monitor_id, query=query2, name=name,
-                                        options={'notify_no_data': True})['id']
-        monitor = dog.Monitor.get(monitor_id)
-        eq(monitor['name'], name)
-        eq(monitor['options']['notify_no_data'], True)
+        monitors = [m for m in dog.Monitor.get_all() if m["id"] == monitor["id"]]
+        assert len(monitors) == 1
 
-        dog.Monitor.delete(monitor_id)
-        resp = dog.Monitor.get(monitor_id)
-        assert 'Monitor not found' in resp.get('errors'), resp
+        assert dog.Monitor.get(monitor["id"])["id"] == monitor["id"]
+        assert dog.Monitor.delete(monitor["id"]) == {"deleted_monitor_id": monitor["id"]}
 
+    def test_monitor_muting(self):
         query1 = "avg(last_1h):sum:system.net.bytes_rcvd{host:host0} > 100"
-        query2 = "avg(last_1h):sum:system.net.bytes_rcvd{host:host0} > 200"
+        query2 = "avg(last_1h):sum:system.net.bytes_rcvd{*} by {host} > 100"
+        monitor1 = dog.Monitor.create(type="metric alert", query=query1)
+        monitor2 = dog.Monitor.create(type="metric alert", query=query2)
 
-        monitor_id1 = dog.Monitor.create(type='metric alert', query=query1)['id']
-        monitor_id2 = dog.Monitor.create(type='metric alert', query=query2)['id']
-        monitors = dog.Monitor.get_all(group_states=['alert', 'warn'])
-        monitor1 = [m for m in monitors if m['id'] == monitor_id1][0]
-        monitor2 = [m for m in monitors if m['id'] == monitor_id2][0]
-        assert monitor1['query'] == query1, monitor1
-        assert monitor2['query'] == query2, monitor2
+        dt = dog.Monitor.mute_all()
+        assert dt["active"] is True
+        assert dt["scope"] == ["*"]
+
+        assert dog.Monitor.unmute_all() is None  # No response expected
+
+        # We shouldn"t be able to mute a simple alert on a scope.
+        assert "errors" in dog.Monitor.mute(monitor1["id"], scope="env:staging")
+
+        monitor2 = dog.Monitor.mute(monitor2["id"], scope="host:foo")
+        assert monitor2["options"]["silenced"] == {"host:foo": None}
+
+        monitor2 = dog.Monitor.unmute(monitor2["id"], scope="host:foo")
+        assert monitor2["options"]["silenced"] == {}
+
+        dog.Monitor.delete(monitor2["id"])
```

Comment:
worth asserting the downtime on this monitor actually is disabled?

Reply:
I don't think so. The monitor CRUD is tested in another test.
I added the deletion of monitor1 though